cypress/integration/ui_spec.js

```diff
@@ -0,0 +1,685 @@
+		.find("[class^=launcher__]")
+		.find("button")
+		.should("be.visible")
+		.click();
+}
+
+function sendMessage(testMessage) {
+	return cy.get("@app")
+		.find("[class^=chat__]")
+		.should("be.visible")
+		.find("[class^=footer__]")
+		.should("be.visible")
+		.find("[class^=text__]")
+		.should("be.visible")
+		.find("textarea")
+		.should("have.focus")
+		.type(`${testMessage}{enter}`);
+}
+
+function findMessage(testMessage) {
+	return cy.get("@app")
+		.find("[class^=wrapper__]")
+		.should("be.visible")
+		.find("[class^=body__]")
+		.should("be.visible")
+		.contains(testMessage)
+		.should("be.visible");
+}
+
+describe("UI", () => {
+	describe("sending messages", () => {
+		beforeEach(() => {
+			cy.visit("/", {
+				onLoad: (window) => {
+					window.initParleyMessenger();
+				},
+			});
+
+			cy.get("[id=app]").as("app");
+		});
+
+		it("should send a new message with the new message showing up in the conversation", () => {
+			const testMessage = `Test message ${Date.now()}`;
+
+			clickOnLauncher();
+			sendMessage(testMessage);
+			findMessage(testMessage);
+		});
+
+		it("should show a generic error when the API returns `status = \"ERROR\"`, but without an error", () => {
+			const testMessage = `Test message ${Date.now()}`;
+
+			cy.intercept("POST", "*/**/messages", {
+				statusCode: 400,
+				body: {status: "ERROR"},
+			});
+
+			clickOnLauncher();
+			sendMessage(testMessage);
+
+			// Validate that api error is visible
+			cy.get("@app")
+				.find("[class^=chat__]")
+				.should("be.visible")
+				.find("[class^=error__]")
+				.should("be.visible")
+				.should("have.text", "The API request failed but the API did not return an error notification");
+		});
+
+		it("should show the `serviceUnreachableNotification` error when the fetch request fails", () => {
+			const testMessage = `Test message ${Date.now()}`;
+
+			cy.intercept("POST", "*/**/messages", {forceNetworkError: true});
+
+			clickOnLauncher();
+			sendMessage(testMessage);
+
+			// Validate that api error is visible
+			cy.get("@app")
+				.find("[class^=chat__]")
+				.should("be.visible")
+				.find("[class^=error__]")
+				.should("be.visible")
+				.should("have.text", "The service is unreachable at the moment, please try again later");
+		});
+
+		it("should show the `messageSendFailed` error when sending a message fails", () => {
+			const testMessage = `Test message ${Date.now()}`;
+
+			cy.intercept("POST", "*/**/messages", {
+				statusCode: 400,
+				body: {
+					status: "ERROR",
+					notifications: [
+						{
+							type: "error",
+							message: "Some specific error",
+						},
+					],
+				},
+			});
+
+			clickOnLauncher();
+			sendMessage(testMessage);
+
+			// Validate that api error is visible
+			cy.get("@app")
+				.find("[class^=chat__]")
+				.should("be.visible")
+				.find("[class^=error__]")
+				.should("be.visible")
+				.should("have.text", "Something went wrong while sending your message, please try again later");
```

Comment:
Moet je ook nog test cases toevoegen voor de andere foutmeldingen (bij pollen messages, en subscribe device). Zie ook https://github.com/parley-messaging/web-library/pull/10/files#diff-d8e59e5be193092a9d8098d65a15d5e619165c3ea9ddc9290389e71925d4b001R118 waarbij ik vermoed dat er nu de verkeerde error wordt terug gegeven als je die raakt.

Reply:
Deze zijn toegevoegd in https://github.com/parley-messaging/web-library/pull/15/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-e6b931217d88546d5c5996678bf3a1bc783ca9f779d2f907052bf3f73de3f377R124 (alpha.3)